src/serializer.rs

```diff
@@ -73,10 +75,10 @@ pub enum DecodeError {
 ///
 /// Serializers that can work with the [`Read`](std::io::Read) &
 /// [`Write`](std::io::Write) traits are generally better suited for lunatic's
-/// ffi, that works on a streaming basis and can avoid unnecessary copies.
+/// FFI, that works on a streaming basis and can avoid unnecessary copies.
 /// Serializer that require raw access to chunks of mutable memories (e.g.
-/// Prost) require additional copies between guest and host memories.
-pub trait Serializer<M> {
+/// `Prost`) require additional copies between guest and host memories.
+pub trait CanSerialize<M> {
```

Comment:
The renaming of `Serializer` to `CanSerialize` is quite interesting.
If I saw a trait with this name, it would lead me to assume it has a method returning a `bool`, but in this case it actually has `endode` and `decode` methods.

I guess the naming update focuses on the trait bound `S: CanSerialize<M>`, but if it couldn't serialize it, then it would not be implemented, which is why I feel like `S: Serializer<M>` works okay.

Is it mostly to avoid confusion with serde's Serializer trait?

Reply:
You are right, the update focuses on the bound.

I think `CanSerialize<M>` is the most descriptive name for this trait. The `M` argument is the important part here. We are not asking if the thing is actually a serializer, but we are asking if it **can serialize** a value `M`. Implementations of it need to provide a `encode` and `decode` message **for M**, if it can.

This leads to much more readable APIs:
```rust
// Instead of
Self::Serializer: Serializer<X>
// we get
Self::Serializer: CanSerialize<X>
```

We know that `Self::Serializer` is a serializer, but we care if it actually can serialize the value `X`. The error messages are also much better:
```
CanSerialize<WeirdType> is not implemented for Bincode.
```
instead of the previous
```
Serializer<WeirdType> is not implemented for Bincode.
```
It's easy to overlook the `<WeirdType>` part and be confused why Bincode is suddenly not a serializer.